client/src/sessions/password.rs

```diff
@@ -1,14 +1,43 @@
 use crate::sessions::Session;
-use tpm2_rs_base::errors::{TssResult, TssTcsError};
+use tpm2_rs_base::errors::{TpmRcResult, TssResult, TssTcsError};
 use tpm2_rs_base::{
     Tpm2bAuth, Tpm2bNonce, Tpm2bSimple, TpmaSession, TpmiShAuthSession, TpmsAuthCommand,
     TpmsAuthResponse,
 };
 
 /// A password session.
+///
+/// # Usage:
+/// ```
+/// use tpm2_rs_client::sessions::PasswordSession;
+///
+/// let password1 = PasswordSession::new("hello world").unwrap(); // from a string
+/// let password2 = PasswordSession::new([1, 2, 3, 4, 5, 6]).unwrap(); // from a byte array
+///
+/// assert_eq!(password1.get_secret(), b"hello world");
+/// assert_eq!(password2.get_secret(), &[1, 2, 3, 4, 5, 6]);
+/// ```
 #[derive(Debug, PartialEq, Default)]
 pub struct PasswordSession {
-    pub auth: Tpm2bAuth,
+    auth: Tpm2bAuth,
+}
+
+impl PasswordSession {
+    /// This function creates a new password session using the
+    /// specified password.
+    ///
+    /// # Errors:
+    /// Returns [`tpm2_rs_base::errors::TpmRcError::Size`] if the
```

Comment:
Bleh, yeah technically wrong - since the error is on the client-side (TPM RC errors can only come from the TPM, not from the client). I am just noticing that I know this is wrong but we don't have an issue open for it yet, so I'll open one.

Anyways, no action on your part, this is currently what we should be doing, but it is technically wrong. I'll open an issue.

Reply:
even more of a reason to add doc test for the error so that when the errorRC gets updated, it produced a failed test so we get to update the doc here.!